### ISSUE TITLE
Add user-friendly application layer with Lightning

### ISSUE THREAD

**Issue by andrrizzi** [andrrizzi/tfep#12]
This should include pre-built normalizing flow models as well as ``LightningModule`` for training.

We could have for example an ``app.models.MAFAffineFlow(torch.nn.Module)`` and ``app.models.CouplingAffineFlow(torch.nn.Module)`` that automatically apply the ZMatrix + Cartesian transformation, and a ``KLTrainedFlow(lightning.LightningModule)`` that takes a flow as input and trains it using KL divergence (uni- and bi-directional).

**Comment by andrrizzi** [comment]
The basic framework for this was implemented in #23, #26, and #28.
